[Email Routing] Add Email Worker guidelines in _index.md (#12591)
cloudflare/cloudflare-docs: content/email-routing/email-workers/_index.md

For example, say that you create an allowlist Email Worker and bind it to a `hello@my-company.com` route. This route will be the email address you share with the world, to make sure that only email addresses on your allowlist are forwarded to your destination address. All other emails will be dropped.

## Guidelines

> It's important to keep the following in mind when writing Email Workers:

Reviewer: ```suggestion
Note the following when writing Email Workers:
```

> 1. Unhandled exceptions will result in temporary SMTP failures, which typically results in the sender retrying later. If you don't want senders retrying, consider adding a top-level `try/catch` block to ensure there are no unhandled exceptions. Some senders may stop sending email to addresses with too many temporary failures.

Reviewer: ```suggestion
1. Unhandled exceptions will result in temporary SMTP failures, which typically result in the sender retrying later. If you do not want senders retrying, add a top-level `try/catch` block to ensure there are no unhandled exceptions. Some senders may stop sending emails to addresses with too many temporary failures.
```

> 2. Email Workers that take longer than 50 seconds to respond will result in a permanent SMTP failure. If some work can be done in the background, consider using [context.waitUntil()](/workers/runtime-apis/handlers/fetch/#contextwaituntil) to execute background tasks after the Email Worker returns. Senders will typically consider permanent SMTP errors to be a bounce, and may stop sending email to addresses which bounce too often. So it's important to not cause unexpected bounces when writing Email Workers.

Reviewer: ```suggestion
2. Email Workers that take longer than 50 seconds to respond will result in a permanent SMTP failure. If some work can be done in the background, consider using [`context.waitUntil()`](/workers/runtime-apis/handlers/fetch/#contextwaituntil) to execute background tasks after the Email Worker returns. Senders will typically consider permanent SMTP errors to be a bounce, and may stop sending emails to addresses which bounce too often. Therefore, it is important to not cause unexpected bounces when writing Email Workers.
```

For example, say that you create an allowlist Email Worker and bind it to a `hello@my-company.com` route. This route will be the email address you share with the world, to make sure that only email addresses on your allowlist are forwarded to your destination address. All other emails will be dropped.

## Guidelines

It's important to keep the following in mind when writing Email Workers:

> 1. Unhandled exceptions will result in a permanent SMTP failures. Email Workers should generally have a top-level `try/catch` block to prevent throwing unhandled exceptions.

Reviewer 2: I'm actually changing this in the next release. Having a top-level try/catch, perhaps even with Sentry, is a good practice. But worker exceptions will result in a temporary smtp error back to the client.
